src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultAdhocWorkspaceFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class DefaultAdhocWorkspaceFactory : AdhocWorkspaceFactory
+    {
+        private readonly HostWorkspaceServicesProvider _hostWorkspaceServicesProvider;
+
+        public DefaultAdhocWorkspaceFactory(HostWorkspaceServicesProvider hostWorkspaceServicesProvider)
+        {
+            if (hostWorkspaceServicesProvider is null)
+            {
+                throw new ArgumentNullException(nameof(hostWorkspaceServicesProvider));
+            }
+
+            _hostWorkspaceServicesProvider = hostWorkspaceServicesProvider;
+        }
+
+        public override AdhocWorkspace Create() => Create(Enumerable.Empty<IWorkspaceService>());
+
+        public override AdhocWorkspace Create(IEnumerable<IWorkspaceService> workspaceServices)
+        {
+            if (workspaceServices is null)
+            {
+                throw new ArgumentNullException(nameof(workspaceServices));
+            }
+
+            var fallbackServices = _hostWorkspaceServicesProvider.GetServices();
```

Comment:
Here's the key piece that VS can hijack to provide the C# workspace's services that is in VS so we don't need to re-pull from MEF